src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -921,6 +921,10 @@ class ClientImpl::Session {
     util::Optional<SubscriptionStore::PendingSubscription> m_pending_flx_sub_set;
     int64_t m_last_sent_flx_query_version = 0;
 
+    util::Optional<IntegrationException> m_client_error;
+    bool m_error_to_send;
```

Comment:
Maybe put `m_error_to_send` up on line 919 so it's with the rest of the session state machine variables.